packages/create-block/lib/init-package-json.js

```diff
@@ -48,6 +71,16 @@ module.exports = async ( {
 	const dependencies = {};
 	const devDependencies = {};
 
+	if ( wpScripts ) {
+		devDependencies[ '@wordpress/scripts' ] =
+			await resolvePackageVersion( '@wordpress/scripts' );
+	}
+
+	if ( wpEnv ) {
+		devDependencies[ '@wordpress/env' ] =
+			await resolvePackageVersion( '@wordpress/env' );
+	}
+
 	if ( npmDependencies && npmDependencies.length ) {
 		for ( const packageArg of npmDependencies ) {
 			try {
```

Comment:
The newly introduced `resolvePackageVersion` is quite powerful. We could use it with other dependencies. Something like that could work nicely:

```diff
- dependencies[ parsed.name ] = parsed.saveSpec || 'latest';
+ dependencies[ parsed.name ] = parsed.saveSpec || resolvePackageVersion( parsed.name );